src/blockchain_db/lmdb/db_lmdb.cpp

```diff
@@ -1516,21 +1516,55 @@ void BlockchainLMDB::update_txpool_tx(const crypto::hash &txid, const txpool_tx_
   }
 }
 
-uint64_t BlockchainLMDB::get_txpool_tx_count() const
+uint64_t BlockchainLMDB::get_txpool_tx_count(bool include_unrelayed_txes) const
 {
   LOG_PRINT_L3("BlockchainLMDB::" << __func__);
   check_open();
 
-  TXN_PREFIX_RDONLY();
   int result;
+  int num_entries = 0;
 
-  MDB_stat db_stats;
-  if ((result = mdb_stat(m_txn, m_txpool_meta, &db_stats)))
-    throw0(DB_ERROR(lmdb_error("Failed to query m_txpool_meta: ", result).c_str()));
+  TXN_PREFIX_RDONLY();
+
+  if (include_unrelayed_txes)
+  {
+    // No filtering, we can get the number of tx the "fast" way
+
+    MDB_stat db_stats;
+    if ((result = mdb_stat(m_txn, m_txpool_meta, &db_stats)))
+      throw0(DB_ERROR(lmdb_error("Failed to query m_txpool_meta: ", result).c_str()));
+    num_entries = db_stats.ms_entries;
+    TXN_POSTFIX_RDONLY();
```

Comment:
duplicated

Reply:
See comment above line 1531. I did not want to degrade performance in case of no filtering; thus the duplication.
Fine to remove. Decision?

Reply:
The TXN_POSTFIX_RDONLY(); call is duplicated. I understand why the code has two branches (and this is good).

Reply:
Oh ok, sorry! Will remove duplicated call indeed :)